src/audio/dai-zephyr.c

```diff
@@ -268,16 +268,55 @@ dai_dma_cb(struct dai_data *dd, struct comp_dev *dev, uint32_t bytes,
 	}
 
 	if (dev->direction == SOF_IPC_STREAM_PLAYBACK) {
+#if CONFIG_IPC_MAJOR_4
+		struct list_item *sink_list;
+		/*
+		 * copy from local buffer to all sinks that are not gateway buffers
```

Comment:
Do we still copy if a sink is inactive or is that for a subsequent update ?

Reply:
In line 304, we verify if sink_dev->state == COMP_STATE_ACTIVE". Additionally, we check if a conversion is assigned for the given sink. Is this what you meant in your question? Thanks.